core/src/main/scala/org/apache/spark/rdd/UnionRDD.scala

```diff
@@ -62,8 +62,14 @@ class UnionRDD[T: ClassTag](
     var rdds: Seq[RDD[T]])
   extends RDD[T](sc, Nil) {  // Nil since we implement getDependencies
 
+  // visible for testing
+  private[spark] val isPartitionEvalParallel: Boolean =
+    rdds.length > conf.getInt("spark.rdd.parallelListingThreshold", 10)
```

Comment:
I agree with @srowen. Let's keep this undocumented because
(1) IDK how many people would actually want to set this (probably not many)
(2) We would need to provide backward compatibility in the future, which makes the code harder to support

The reason why it's even a config is only for safety. In the unlikely event that this somehow screws things up badly we can always tell the user to just set this to 1 to revert to the original behvaior. Normally, however, we wouldn't recommend the user to even touch this at all.